mle/integration/local_git.py

```diff
@@ -0,0 +1,110 @@
+from git import Repo, NULL_TREE
+from datetime import datetime
+
+
+class GitIntegration:
+    def __init__(self, path):
+        self.repo_path = path
+        self.repo = Repo(self.repo_path)
+        if self.repo.bare:
+            raise Exception("Repository is not valid or is bare.")
+
+    def get_repo_status(self):
+        """
+        Get the status of a git repository
+        :return: List of changed files
+        """
+        try:
+            changed_files = []
+            for diff in self.repo.index.diff(None):
+                changed_files.append({
+                    'file_path': diff.a_path,
+                    'change_type': diff.change_type,
+                    'author': self.repo.head.commit.author.name,
+                    'date': datetime.fromtimestamp(self.repo.head.commit.committed_date).strftime("%Y-%m-%d %H:%M:%S")
+                })
+
+            return changed_files
+
+        except Exception as e:
+            return f"An error occurred: {str(e)}"
+
+    def get_commit_history(self, limit=None):
```

Comment:
so this function gets all history

will you provide a function to get 1 week/day/month history?

Reply:
Good suggestion, I can add the date as the parameter

Reply:
Or I can add a new function to aggregate all the metadata and put the date range into that function